Source/RuntimeImageLoader/Private/Texture2DAnimation/GIFTexture.cpp

```diff
@@ -0,0 +1,33 @@
+// Copyright 2023 Peter Leontev and Muhammad Ahmed Saleem. All Rights Reserved.
+
+#include "Texture2DAnimation/GIFTexture.h"
+#include "RuntimeImageLoaderLog.h"
+
+DEFINE_LOG_CATEGORY(GifTexture);
+
+UAnimatedTexture2D* UGIFTexture::Init(const FString& GIFFilename)
+{
+	TUniquePtr<FRuntimeGIFLoaderHelper> Decoder = MakeUnique<FRuntimeGIFLoaderHelper>();
+
+	if (Decoder)
+	{
+		Decoder->GIFDecoding(TCHAR_TO_UTF8(*GIFFilename));
+		
```

Comment:
Make sure GIF decoding is done in a separate thread (as per our Discord conversation). 

Look into Async(...), AsyncTask(...) or AsyncThread(...)

Reply:
We don't want to make Game Thread wait until GIF decoding is done